src/schema/notifications.ts

```diff
@@ -214,6 +288,150 @@ export class ZimbraNotifications {
 		this.batchProcessItems(modifiedItems, this.processTagsNotifications);
 	};
 
+	private processAppointmentNotifications = (items: any) => {
+		items.forEach((i: any) => {
+			const item = normalizeCalendarItem(i);
+			if (item.invitations && item.invitations.length) {
+				const nextAlarm = get(item, 'nextAlarm');
+				const invitation = get(item, 'invitations.0');
+				const { date, folderId } = item;
+
+				const invitationComponent = get(invitation, 'components.0');
+				const {
+					name,
+					allDay = false, // presents only if it's true
+					location,
+					inviteId,
+					status: participationStatus,
+					freeBusy,
+					freeBusyActual,
+					class: classname,
+					rsvp: otherAttendees,
+					recurrence: isRecurring = false, // presents only if it's recurring event
+					organizer: { name: organizerName, address }
+				} = invitationComponent;
+
+				// Get utc info from invitation.
+				let eventStartUtc: number;
+				let eventEndUtc: number;
+				let duration: number;
+
+				if (allDay) {
+					// In case of allDay event, transform `20200401` to `2020-04-01`
+					// and convert it to utc
+					const startDate = get(invitationComponent, 'start.0.date');
+					const endDate = get(invitationComponent, 'end.0.date');
+
+					eventStartUtc = utcFromDateString(startDate);
```

Comment:
can we achieve this using moment?

Reply:
We are not using moment in this repo. Shall we install moment?

I had refused to install because it will be used at very few places.